lib/src/validator/git_status.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:path/path.dart' as p;
+
+import '../git.dart' as git;
+import '../log.dart' as log;
+import '../utils.dart';
+import '../validator.dart';
+
+/// A validator that validates that no checked in files are modiofied in git.
+///
+/// Doesn't report on newly added files, as generated files might not be checked
+/// in to git.
+class GitStatusValidator extends Validator {
+  @override
+  Future<void> validate() async {
+    if (package.inGitRepo) {
```

Comment:
nit: please return early instead of nesting a bit more :rofl: 
```suggestion
    if (package.inGitRepo) {
      return;
    }
```

Reply:
Done